### PR TITLE
Add menu for restaurant

### PR DESCRIPTION
**Pull request by ceca2017**
### Add menu for restaurant
https://www.pivotaltracker.com/story/show/145927849

- Add feature for restaurant owner to create his menu pages (later to be used to add dishes to).
- Add restaurant owner to the restaurant model
- Update the tests to make them pass with a restaurant owner

### CODE REVIEW

**Comment by ceca2017** [comment]
Thanks for the verbal comments Thomas earlier today!
Here are some of our comments:

app/controllers/menus_controller.rb:
We refactored the code to search via find, however the @user and @restaurant seems to be needed for the path for the form in the view to work.
Not sure if we can sanitise the params, as we are only looking for one specific param at the time.

app/controllers/restaurants_controller.rb
On row 22, we need the user.id in @restaurant.user_id = current_user.id for the assignment to work

app/controllers/user/restaurants_controller.rb
The reason for the index and show methods is that the corresponding views are for the restaurant owner only.

config/routes.rb
We took out the get statements and instead added an owners namespace. We then had to refactor the code to have owners instead of user (a bit of work).

Also we want to thank Raoul and Andy for their support!

File: app/controllers/owners/restaurants_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Owners::RestaurantsController < ApplicationController
+
+  def index
+    @owner = current_user.id
```
Comment (diraulo) [inline app/controllers/owners/restaurants_controller.rb]: okay since you need `@owner` in both actions you can do the following

1. Create a private method
```ruby
def owner
  @owner = User.find(current_user.id)
end
```

2. Add a `before_action`

```ruby
before_action :owner
```

What that means is that you can get rid of line 4 and line 10 in this controller.

I'm guessing you're using `@owner_id` in the view, after you make the above change, go to the view and use `@owner.id`

File: app/controllers/owners/menus_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class Owners::MenusController < ApplicationController
+  def new
+    @menu = Menu.new
+    @restaurant = Restaurant.find(params[:restaurant_id])
+  end
+
+  def create
+    @menu = Menu.new(name: params[:menu][:name], restaurant_id: params[:restaurant_id])
```
Comment (tochman) [inline app/controllers/owners/menus_controller.rb]: I still don't see sanitized params.

File: app/controllers/owners/restaurants_controller.rb (hunk 1 above)
Comment (tochman) [inline app/controllers/owners/restaurants_controller.rb]: If you only use the variable in the method you can use a local variable.

File: app/controllers/owners/restaurants_controller.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    @restaurants = Restaurant.where(user_id: @owner)
+  end
+
+  def show
+    @restaurant_id = params[:id]
+    @owner_id = current_user.id
```
Comment (tochman) [inline app/controllers/owners/restaurants_controller.rb]: What is this variable used for? Why only store the id?

File: app/controllers/restaurants_controller.rb [hunk 4]
```diff
@@ -19,6 +19,7 @@ def new
 
   def create
     @restaurant = Restaurant.new(restaurant_params)
+    @restaurant.user_id = current_user.id
```
Comment (diraulo) [inline app/controllers/restaurants_controller.rb]: Here I think you can do `@restaurant = current_user.restaurants.build(restaurant_params)` not sure but put a binding.pry and try that out.

Comment (tochman) [inline app/controllers/restaurants_controller.rb]: I would set the user in a more ActiveRecordish way: 
```
@restaurant.user = current_user
```

File: features/create_menu.feature [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  In order to categorise the dishes
+  I would like to be able to create my own menus
+
+    Background:
+      Given "o.dania@aol.com" exists and has 2 restaurants belonging to 2 restaurant categories
+      And I am logged in as "o.dania@aol.com"
+
+    Scenario: Add Menu on Restaurant Owner Page
+      Given I click on link "View My Restaurant Page"
+      Then I should visit "o.dania@aol.com" restaurants page
```
Comment (tochman) [inline features/create_menu.feature]: Please revisit this step. It looks like an assertion but is it?

File: app/controllers/owners/restaurants_controller.rb (hunk 1 above)
Comment (diraulo) [reply app/controllers/owners/restaurants_controller.rb]: After going through all your code, you're actually not using `@owner_id` anywhere so get rid of it and the before filter can now become

```ruby
before_action :owner, only: :index
```

File: features/step_definitions/create_menu_steps.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+Given(/^I should visit "([^"]*)" restaurants page$/) do |email|
```
Comment (tochman) [inline features/step_definitions/create_menu_steps.rb]: I would rephrase this to `Given I visit...`

**Review by tochman — DISMISSED** [review]

**Review by diraulo — CHANGES_REQUESTED** [review]
😒

File: app/controllers/owners/menus_controller.rb (hunk 2 above)
Comment (ceca2017) [reply app/controllers/owners/menus_controller.rb]: We updated the code with sanitising the params. However, is this needed as we in the previous code (above) only are looking for a specific params attribute?

File: app/controllers/owners/restaurants_controller.rb (hunk 1 above)
Comment (ceca2017) [reply app/controllers/owners/restaurants_controller.rb]: Yes, done. Thanks

File: app/controllers/owners/restaurants_controller.rb (hunk 3 above)
Comment (ceca2017) [reply app/controllers/owners/restaurants_controller.rb]: It is not needed. It is now gone :)

File: app/controllers/restaurants_controller.rb (hunk 4 above)
Comment (ceca2017) [reply app/controllers/restaurants_controller.rb]: Done

File: features/create_menu.feature (hunk 5 above)
Comment (ceca2017) [reply features/create_menu.feature]: You are right. We updated the code.

File: features/step_definitions/create_menu_steps.rb (hunk 6 above)
Comment (ceca2017) [reply features/step_definitions/create_menu_steps.rb]: Yup. Done

File: app/controllers/owners/restaurants_controller.rb (hunk 1 above)
Comment (ceca2017) [reply app/controllers/owners/restaurants_controller.rb]: We made updates to the code when doing the review of Thomas comments. Hopefully this solution will work for your comments as well.

File: app/controllers/restaurants_controller.rb (hunk 4 above)
Comment (ceca2017) [reply app/controllers/restaurants_controller.rb]: Thanks Raoul! Also updated with the comments from Thomas. Hopefully this will work for your comment as well.

**Comment by ceca2017** [comment]
Thanks to you both Thomas and Raoul for your reviews! Hopefully the new commit will address your comments.

**Review by diraulo — APPROVED** [review]
:shipit:

**Comment by diraulo** [comment]
@tochman Any other objections on this?

**Comment by tochman** [comment]
Sure @diraulo No drama.
